lib/internal/Magento/Framework/Encryption/Adapter/Mcrypt.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Magento\Framework\Encryption\Adapter;
+
+class Mcrypt implements EncryptionAdapterInterface
+{
+    /**
+     * @var string
+     */
+    private $cipher;
+
+    /**
+     * @var string
+     */
+    private $mode;
+
+    /**
+     * @var string
+     */
+    private $initVector;
+
+    /**
+     * @var resource
+     */
+    private $handle;
+
+    /**
+     * Mcrypt constructor.
+     * @param string $key
+     * @param string $cipher
+     * @param string $mode
+     * @param bool $initVector
```

Comment:
According to the method implementation, `$initVector` could be `true`, `false` or `string`, which does not match the annotation. Also having mixed types for arguments is complicating the logic and is confusing.